### ISSUE TITLE
We should host our own httpbin server for HttpClient tests

### ISSUE THREAD

**Issue by geoffkizer** [dotnet/runtime#47351]
Currently we are using the public server at http://httpbin.org for some decompression tests, see here: https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Decompression.cs#L206

Rather than just use the public server, we really should host our own httpbin server in Azure or similar. This should be easy to do.

**Comment by ManickaP** [comment]
We already have remote endpoints running on Azure: https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/System/Net/Configuration.Http.cs
Including for decompression: https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/System/Net/Configuration.Http.cs#L65
But according to the comment we're not doing it right:
> // The remote server endpoint was written to use DeflateStream, which isn't actually a correct
   // implementation of the deflate protocol (the deflate protocol requires the zlib wrapper around
   // deflate).  Until we can get that updated (and deal with previous releases still testing it
   // via a DeflateStream-based implementation), we utilize httpbin.org to help validate behavior.

So this might be just about fixing that endpoint.

Also related: https://github.com/dotnet/runtime/issues/40380

**Comment by geoffkizer** [comment]
If we fixed the endpoint, then we could remove the use of httpbin.

That said, httpbin is a pretty cool little component. I'd kind of like to see us use this more places. It has more capabilities than our existing remote endpoint and fewer issues (e.g. it supports deflate correctly). In the long run we could replace our remote endpoint with httpbin entirely, I think.

**Comment by ManickaP** [comment]
> In the long run we could replace our remote endpoint with httpbin entirely, I think.

Ohh, now I get it, that's a brilliant idea :+1:

**Comment by wfurt** [comment]
we could use Docker as we do for some tests. And possibly use same image hosted in Azure. We have parametrization for tests via configuration and it would be cool to be able to run all tests tests from places with limited connectivity (like bus when that is back In business)

**Comment by geoffkizer** [comment]
> we could use Docker as we do for some tests

Yeah, this would be great to do as well. The httpbin stuff is docker-hosted already as i understand it.

**Comment by karelz** [comment]
Triage: Primarily set up public endpoint (Docker image?) in Azure instead of hitting public URL http://httpbin.org
